### PR TITLE
feat: introduce block variable support to more elements

### PR DESCRIPTION
**Pull request by neatbyte-vnobis**
## Changes
Implements issue "Introduce block variable support to all other elements".

## How Has This Been Tested?
Manual

## Documentation
None

### CODE REVIEW

**Comment by SvenAlHamad** [comment]
Hey @neatbyte-vnobis - on this one, I think I might have missed one item in my specification, and that is the variable support for the icon element.

**Comment by neatbyte-vnobis** [comment]
@SvenAlHamad 
This PR was updated and icon variable support was added (this includes some changes to editor icon component to support it).

Important thing: I have checked the page preview screen and fond out that embed elements are not displaying correctly, now it is fixed.
Also I have added required validation (placeholder, description, and url fields validation), it was missing and it could cause issues without it.

**Comment by neatbyte-vnobis** [comment]
@SvenAlHamad @adrians5j 
Any updates on this PR?
Maybe it worth to merge latest `next` into it and retest it

**Review by adrians5j — CHANGES_REQUESTED** [review]
Did a quick manual test. Some notes below ⬇️
 

1. Scratched 1st note. Sorry.

2. When I tried to test Twitter, I received the following err message after I dropped the block onto a new page ⬇️ 
image

BTW can confirm the same happened when I tried YT.

File: packages/api-page-builder/src/graphql/elementProcessors/embed.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default new ContextPlugin<PbContext>(context => {
         const variables = useElementVariables(block, element);
         const value = variables?.length > 0 ? variables[0].value : null;
 
-        if (value !== null) {
+        if (value !== undefined && value !== null) {
```
Comment (adrians5j) [inline packages/api-page-builder/src/graphql/elementProcessors/embed.ts:18]: Why not just do `typeof value === 'string'` here? 🤔

**Review by adrians5j — CHANGES_REQUESTED** [review]
One question here.

**Pull request merged**